File: Programming_2/C++/lab_5/lab_5.2/Numbers.cpp
//
// Created by Danil on 24.03.2020.
//

#include "Numbers.h"

Numbers::Numbers(int size, const char *array) : Rows(size, array){}

void Numbers::extend() {
    int new_size(size*2);
    char *new_array = new char[new_size + 1];

    for(int i = 0, j = 0; i < size; i++, j+=2)
    {
        new_array[j] = new_array[j+1] = chars[i];
    }
    size = new_size;
    memcpy(chars, new_array, size);
    chars[size] = '\0';
    delete[] new_array;
}

File: Programming_2/C#/lab_5/lab_5.2/Numbers.cs
namespace lab_5_2
{
    public class Numbers : Rows
    { 
        public Numbers(params char[] numbers) : base(numbers)
        {}
        
        public override void Enlargement()
        {
            char[] chars = new char[Row.Length*2];
            for (int i = 0, j = 0; i < Row.Length; i++, j+=2)
            {
                chars[j] = chars[j+1] = Row[i];
            }
            Row = chars;
        }
    }
}